src/decorator.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+import * as _helpers from './helpers';
+import * as protos from '../protos/protos';
+
+const enhancedTypesJson = require('./enhanced-types.json');
+
+// Walk the tree of nested namespaces contained within the enhanced-types.json file
+function walkNamespaces(
+  // tslint:disable-next-line no-any
+  jsonNode: Record<string, any>,
+  // tslint:disable-next-line no-any
+  rootNamespace: Record<string, any>
+): void {
+  for (const namespaceName in jsonNode) {
+    if (Object.hasOwnProperty.call(jsonNode, namespaceName)) {
+      const namespace =
+        namespaceName in rootNamespace
+          ? rootNamespace[namespaceName]
+          : undefined;
```

Comment:
You don't have to use a ternary here, `rootNamespace[namespaceName]` will be `undefined` if it's not there, so just `const namespace = rootNamespace[namespaceName]` is equivalent to this code

Reply:
Done.